### ISSUE TITLE
Forced interrupt causes panic

### ISSUE THREAD

**Issue by taylorflatt** [taylorflatt/go-chat#2]
Sometimes when force interrupting (ctrl+c), the entire server crashes.

Server:
```
2017/03/18 18:22:11 [UnRegister]: Unregistering client user1
2017/03/18 18:22:11 [removeClient]: Removed client user1
2017/03/18 18:22:11 [removeClientFromGroup]: Removed client user1 from group1
2017/03/18 18:22:11 [user2 user3]
2017/03/18 18:22:11 transport: http2Server.HandleStreams failed to read frame: read tcp [::1]:12021->[::1]:58082: wsarecv: An existing connection was forcibly closed by the remote host.
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x8 pc=0x403e33]

goroutine 20 [running]:
panic(0x78ff80, 0xc04200c080)
        C:/Go/src/runtime/panic.go:500 +0x1af
main.listenToClient(0x9be680, 0xc042168070, 0xc0421381e0)
        C:/Users/Taylor/Work/src/github.com/taylorflatt/go-chat/Server/server.go:310 +0x93
created by main.(*server).RouteChat
        C:/Users/Taylor/Work/src/github.com/taylorflatt/go-chat/Server/server.go:265 +0x1d2
exit status 2
```

The first 4 lines are expected when a user forcibly exits the session. But the panic is not. This was experienced when three clients were in a group chat.

Note: This problem doesn't seem to persist in a dual-client session. Nor is it a problem when not in a group. 

Client: 
```
2017/03/18 18:22:11 transport: http2Client.notifyError got notified that the client transport was broken read tcp [::1]:58087->[::1]:12021: wsarecv: An existing connection was forcibly closed by the remote host..
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x4045c2]

goroutine 21 [running]:
panic(0x78d5a0, 0xc04200c080)
        C:/Go/src/runtime/panic.go:500 +0x1af
main.receiveMessages(0x9a9c60, 0xc0421465a0, 0xc0421407e0, 0xc042146458, 0x6)
        C:/Users/Taylor/Work/src/github.com/taylorflatt/go-chat/Client/client.go:332 +0x52
created by main.main
        C:/Users/Taylor/Work/src/github.com/taylorflatt/go-chat/Client/client.go:295 +0x22f3
exit status 2
```

**Comment by taylorflatt** [comment]
I narrowed the issue down. The panic only occurs when users are in a group and have sent messages (the buffer is non-empty).

The place in code the panic points to is in reference to a log statement which includes the following: 

```
log.Printf("[broadcast] Client " + msg.Sender + " sent " + msg.Receiver 
         + " a message: " + msg.Message)
```

**Comment by taylorflatt** [comment]
This has been fixed in #4.
